modules/shapefile/src/lib/parsers/parse-shp.js

```diff
@@ -19,7 +19,7 @@ const STATE = {
 
 class SHPParser {
   constructor() {
-    this.binaryReader = new BinaryChunkReader();
+    this.binaryReader = new BinaryChunkReader({maxRewindBytes: SHP_RECORD_HEADER_SIZE});
```

Comment:
I guess I'd just forgotten to set this. This is necessary so that the `BinaryChunkReader` doesn't dispose of `ArrayBuffers` prematurely when a `rewind` might be needed in the future.

This assumes that 12 bytes (the SHP record header size) is the max that will ever need to be rewound.

Reply:
Makes sense. Actually, I think you-know-who's logic just rewinds 8 bytes at a time, moving one 32 bit word forward at a time until it finds a block of 12 bytes that looks like a valid SHP record header.